client/my-sites/plan-features-2023-grid/billing-timeframe.tsx

```diff
@@ -3,54 +3,67 @@ import {
 	isWpcomEnterpriseGridPlan,
 	PLAN_BIENNIAL_PERIOD,
 	PLAN_ANNUAL_PERIOD,
+	PlanSlug,
+	getPlanSlugForTermVariant,
+	TERM_ANNUALLY,
 } from '@automattic/calypso-products';
 import { formatCurrency } from '@automattic/format-currency';
 import { localize, TranslateResult, useTranslate } from 'i18n-calypso';
 import { FunctionComponent } from 'react';
 import { useSelector } from 'react-redux';
+import usePlanPrices from 'calypso/my-sites/plans/hooks/use-plan-prices';
 import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
 
 interface Props {
 	planName: string;
 	billingTimeframe: TranslateResult;
 	billingPeriod: number | null | undefined;
-	rawPrice: number | null;
-	maybeDiscountedFullTermPrice: number | null;
-	annualPricePerMonth: number | null;
 	isMonthlyPlan: boolean;
 }
 
 function usePerMonthDescription( {
-	rawPrice,
-	maybeDiscountedFullTermPrice,
-	annualPricePerMonth,
 	isMonthlyPlan,
 	planName,
 	billingPeriod,
 }: Omit< Props, 'billingTimeframe' > ) {
 	const translate = useTranslate();
 	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const planPrices = usePlanPrices( {
+		planSlug: planName as PlanSlug,
+		returnMonthly: isMonthlyPlan,
+	} );
+	const planYearlyVariantPrices = usePlanPrices( {
+		planSlug:
+			getPlanSlugForTermVariant( planName as PlanSlug, TERM_ANNUALLY ) ?? ( '' as PlanSlug ),
+		returnMonthly: true,
+	} );
+	const maybeDiscountedPrice =
+		planPrices.planDiscountedRawPrice || planPrices.discountedRawPrice || planPrices.rawPrice;
+	// we want the raw price (discounted or not) for the yearly variant, not the site-plan discounted one
```

Comment:
Can you please explain this change?

Reply:
Thanks, @aneeshd16. The price derivation was transferred from [here](https://github.com/Automattic/wp-calypso/pull/75073/files#diff-e268185bf1c246017f6796f5f74ff3c0ae49b12e2acf525d8074859449573a19L1056), hopefully unaltered. I believe the comment may have been an earlier understanding of what happens there, that it doesn't reach into `planDiscountedRawPrice` (unless I miss something from the original). I guess it's misleading, as that's not the actual difference between `planDiscountedRawPrice` and `discountedRawPrice`, right? 

Let's update to something more clear. I think some comment is warranted, even if it just triggers plain curiosity. Any ideas?

Reply:
Maybe something like: (please feel free to reword 😅 )
```
We want `yearlyVariantMaybeDiscountedPricePerMonth` to be the raw price the user would pay if they choose an annual plan instead of the monthly one. So pro-rated credits should not be taken into account. 
```

My understanding of the "raw" terminology is that it's the numerical value of the price (for example: `100` instead of `$100.00`). Please correct me if this is wrong!

Reply:
Happy with what you wrote above.

> My understanding of the "raw" terminology is that it's the numerical value of the price (for example: 100 instead of $100.00). Please correct me if this is wrong!

I think that's pretty much it. Both selectors will calculate something over `raw_price` (one considering a site discounted plan, the other just the plan) - if I understand correctly their internals.